core/toolbox.js

```diff
@@ -588,6 +589,83 @@ Blockly.Toolbox.TreeNode.prototype.onDoubleClick_ = function(e) {
 };
 
 /**
+ * Modify original onKeyDown handler to swap
+ * LEFT/RIGHT keys with UP/DOWN in horizontalLayout.
+ * @param {!goog.events.BrowserEvent} e The browser event.
+ * @return {boolean} The handled value.
+ * @override
+ * @private
+ */
+Blockly.Toolbox.TreeNode.prototype.onKeyDown = function(e) {
+  var handled = true;
+  var vertical = !this.horizontalLayout_;
+
+  var right = vertical ? 'RIGHT' : 'DOWN';
```

Comment:
var right = vertical ? goog.events.KeyCodes.RIGHT : goog.events.KeyCodes.DOWN

and the same for the next few lines.  Then use the variables directly, instead of doing lookup with the variables.